tests/integration/synapseclient/models/async/test_wiki_async.py

```diff
@@ -0,0 +1,751 @@
+        schedule_for_cleanup: Callable[..., None],
+    ) -> None:
+        """Test getting attachment handles for a gz file."""
+        # GIVEN a wiki page with a gz attachment
+        wiki_page, attachment_name = wiki_page_with_gz_attachment
+        # WHEN getting attachment handles
+        attachment_handles = await wiki_page.get_attachment_handles_async(
+            synapse_client=self.syn
+        )
+
+        # THEN attachment handles should be returned
+        assert len(attachment_handles["list"]) > 0
+        # Verify the attachment name contains .gz
+        assert any(
+            handle.get("fileName", "").endswith(".gz")
+            for handle in attachment_handles["list"]
+        )
+        schedule_for_cleanup(attachment_handles)
+
+    async def test_get_attachment_url_gz_file(
+        self,
+        wiki_page_with_gz_attachment: tuple[WikiPage, str],
+        schedule_for_cleanup: Callable[..., None],
+    ) -> None:
+        """Test getting attachment URL for a gz file."""
+        # GIVEN a wiki page with a gz attachment
+        wiki_page, attachment_name = wiki_page_with_gz_attachment
+
+        # WHEN getting attachment URL
+        attachment_url = await wiki_page.get_attachment_async(
+            file_name=attachment_name, download_file=False, synapse_client=self.syn
+        )
+        # THEN a URL should be returned
+        assert len(attachment_url) > 0
+        schedule_for_cleanup(attachment_url)
+
+    async def test_download_attachment_gz_file(
+        self,
+        wiki_page_with_gz_attachment: tuple[WikiPage, str],
+        schedule_for_cleanup: Callable[..., None],
+    ) -> None:
+        """Test downloading a gz attachment file."""
+        # GIVEN a wiki page with a gz attachment
+        wiki_page, attachment_name = wiki_page_with_gz_attachment
+
+        # AND a download location
+        download_dir = tempfile.mkdtemp()
+        schedule_for_cleanup(download_dir)
+
+        # WHEN downloading the gz attachment
+        downloaded_path = await wiki_page.get_attachment_async(
+            file_name=attachment_name,
+            download_file=True,
+            download_location=download_dir,
+            synapse_client=self.syn,
+        )
+        schedule_for_cleanup(downloaded_path)
+
+        # THEN the file should be downloaded
+        assert os.path.exists(downloaded_path)
+        assert os.path.basename(downloaded_path) + ".gz" == attachment_name
+
+    async def test_get_attachment_preview_url_gz_file(
+        self,
+        wiki_page_with_gz_attachment: tuple[WikiPage, str],
+        schedule_for_cleanup: Callable[..., None],
+    ) -> None:
+        """Test getting attachment preview URL for a gz file."""
+        # GIVEN a wiki page with a gz attachment
+        wiki_page, attachment_name = wiki_page_with_gz_attachment
+        # Sleep for 0.5 minutes to ensure the attachment preview is created
+        time.sleep(0.5 * 60)
```

Comment:
When the test is in an asyncio context please use:

`await asyncio.sleep(0.5 * 60)` instead.